File: ui/src/domain/job.ts
export type JobStatus = 'queued' | 'running' | 'succeeded' | 'failed'

export type JobType = 'backup' | 'restore'

export interface Job {
  id: string
  type: JobType
  status: JobStatus
  error?: string
  backup_id?: string
  created_at: string
  started_at?: string
  finished_at?: string
}

export function isActiveJob(job: Job): boolean {
  return job.status === 'queued' || job.status === 'running'
}

File: internal/domain/job.go
package domain

import "time"

type JobStatus string

const (
	JobStatusQueued    JobStatus = "queued"
	JobStatusRunning   JobStatus = "running"
	JobStatusSucceeded JobStatus = "succeeded"
	JobStatusFailed    JobStatus = "failed"
)

type JobType string

const (
	JobTypeBackup  JobType = "backup"
	JobTypeRestore JobType = "restore"
)

// Job tracks an asynchronous backup or restore operation.
type Job struct {
	ID         string     `json:"id"`
	Type       JobType    `json:"type"`
	Status     JobStatus  `json:"status"`
	Error      string     `json:"error,omitempty"`
	BackupID   string     `json:"backup_id,omitempty"`
	CreatedAt  time.Time  `json:"created_at"`
	StartedAt  *time.Time `json:"started_at,omitempty"`
	FinishedAt *time.Time `json:"finished_at,omitempty"`
}
